pkg/compute/init.go

```diff
@@ -4,32 +4,59 @@ import (
 	"crypto/rand"
 	"fmt"
 	"io"
+	mathRand "math/rand"
+	"net"
 	"os"
 	"path/filepath"
 	"regexp"
+	"strconv"
+	"strings"
 	"time"
 
+	"github.com/dustinkirkland/golang-petname"
 	"github.com/fastly/cli/pkg/common"
 	"github.com/fastly/cli/pkg/compute/manifest"
 	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
 	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/fastly"
 	"gopkg.in/src-d/go-git.v4"
 )
 
-const defaultTemplate = "https://github.com/fastly/fastly-template-rust-default"
+type template struct {
+	Name string
+	Path string
+}
+
+const (
+	defaultTemplate       = "https://github.com/fastly/fastly-template-rust-default.git"
+	defaultTopLevelDomain = "edgecompute.app"
+	manageServiceBaseURL  = "https://manage.fastly.com/configure/services/"
+)
 
 var (
+	gitRepositoryRegEx        = regexp.MustCompile(`((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?`)
+	domainNameRegEx           = regexp.MustCompile(`(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]`)
```

Comment:
I am afraid this is incomplete, but I don't have the strength/knowledge/intelligence to test it. Does it come from some known source? Is it possible to be abused?

Reply:
It seemed the most robust in the top SO response for this question, however I agree with you that it could be brittle and I'm sad the Go standard library doesn't offer something such as how you can parse and validate IP addresses.